### ISSUE TITLE
shared-library versioning

### ISSUE THREAD

**Issue by stevengj** [JuliaLang/utf8proc#24]
We should really include a version number in the shared library (this is different from the utf8proc source version: the source version indicates API compatibility, while the library version encodes ABI compatibility).     I have used [libtool for this](http://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html) in the past, but the whole autotools stack is probably not worth the trouble here (although we could use just the `libtool` script, which could be bundled by `make dist`).

**Comment by stevengj** [comment]
See also the [discussion of a soname in the Fedora package](https://bugzilla.redhat.com/show_bug.cgi?id=1058019).   cc: @nalimilan

**Comment by nalimilan** [comment]
Great! Note that by choosing 1.2.0, you signal an ABI break as compared with the 0.1 SOVERSION I have used in the Fedora package. Is that intended?

**Comment by stevengj** [comment]
Yes, that is intended.   The ABI has changed incompatibly because a `struct` was changed by #20 (although the API is backwards compatible).

Of course, since this is the first release with an soname, I could have chosen `1.0.0` or even `0.0.0` (though Darwin doesn't like zero versions), but I figured it was less confusing to start with a version that matches the API version (even if they subsequently diverge).
